File: src/main/java/io/github/faecraft/heartshapedbox/body/BuiltInParts.java
package io.github.faecraft.heartshapedbox.body;

import io.github.faecraft.heartshapedbox.body.impl.ArmBodyPart;
import io.github.faecraft.heartshapedbox.body.impl.FootBodyPart;
import io.github.faecraft.heartshapedbox.body.impl.LegBodyPart;
import net.minecraft.util.Identifier;
import net.minecraft.util.Pair;

import static io.github.faecraft.heartshapedbox.body.BodyPartSide.LEFT;
import static io.github.faecraft.heartshapedbox.body.BodyPartSide.RIGHT;
import static io.github.faecraft.heartshapedbox.main.HSBMain.MOD_ID;

public class BuiltInParts {
    // Head
    public static final Identifier HEAD = new Identifier(MOD_ID, "head");
    
    // Arms
    public static final Identifier LEFT_ARM = new Identifier(MOD_ID, "arm/" + LEFT.lowerName());
    public static final Identifier RIGHT_ARM = new Identifier(MOD_ID, "arm/" + RIGHT.lowerName());
    
    // Legs
    public static final Identifier LEFT_LEG = new Identifier(MOD_ID, "leg/" + LEFT.lowerName());
    public static final Identifier RIGHT_LEG = new Identifier(MOD_ID, "leg/" + RIGHT.lowerName());
    
    // Feet
    public static final Identifier LEFT_FOOT = new Identifier(MOD_ID, "foot/" + LEFT.lowerName());
    public static final Identifier RIGHT_FOOT = new Identifier(MOD_ID, "foot/" + RIGHT.lowerName());
    
    
    // Pairs
    
    public static Pair<ArmBodyPart, ArmBodyPart> getArms(BodyPartProvider provider) {
        return new Pair<>(
            (ArmBodyPart)provider.getOrThrow(BuiltInParts.LEFT_ARM),
            (ArmBodyPart)provider.getOrThrow(BuiltInParts.RIGHT_ARM)
        );
    }
    
    public static Pair<LegBodyPart, LegBodyPart> getLegs(BodyPartProvider provider) {
        return new Pair<>(
            (LegBodyPart)provider.getOrThrow(BuiltInParts.LEFT_LEG),
            (LegBodyPart)provider.getOrThrow(BuiltInParts.RIGHT_LEG)
        );
    }
    
    public static Pair<FootBodyPart, FootBodyPart> getFeet(BodyPartProvider provider) {
        return new Pair<>(
            (FootBodyPart)provider.getOrThrow(BuiltInParts.LEFT_FOOT),
            (FootBodyPart)provider.getOrThrow(BuiltInParts.RIGHT_FOOT)
        );
    }
}

File: src/main/kotlin/io/github/faecraft/heartshapedbox/body/BuiltInParts.kt
package io.github.faecraft.heartshapedbox.body

import io.github.faecraft.heartshapedbox.body.impl.ArmBodyPart
import io.github.faecraft.heartshapedbox.body.impl.FootBodyPart
import io.github.faecraft.heartshapedbox.body.impl.LegBodyPart
import io.github.faecraft.heartshapedbox.main.HSBMain
import net.minecraft.util.Identifier
import net.minecraft.util.Pair

public object BuiltInParts {
    // Head
    @JvmField
    public val HEAD: Identifier = Identifier(HSBMain.MOD_ID, "head")

    // Arms
    @JvmField
    public val LEFT_ARM: Identifier = Identifier(HSBMain.MOD_ID, "arm/${BodyPartSide.LEFT.lowerName()}")

    @JvmField
    public val RIGHT_ARM: Identifier = Identifier(HSBMain.MOD_ID, "arm/${BodyPartSide.RIGHT.lowerName()}")

    // Legs
    @JvmField
    public val LEFT_LEG: Identifier = Identifier(HSBMain.MOD_ID, "leg/${BodyPartSide.LEFT.lowerName()}")

    @JvmField
    public val RIGHT_LEG: Identifier = Identifier(HSBMain.MOD_ID, "leg/${BodyPartSide.RIGHT.lowerName()}")

    // Feet
    @JvmField
    public val LEFT_FOOT: Identifier = Identifier(HSBMain.MOD_ID, "foot/${BodyPartSide.LEFT.lowerName()}")

    @JvmField
    public val RIGHT_FOOT: Identifier = Identifier(HSBMain.MOD_ID, "foot/${BodyPartSide.RIGHT.lowerName()}")

    // Pairs
    public fun getArms(provider: BodyPartProvider): Pair<ArmBodyPart, ArmBodyPart> {
        return Pair(
            provider.getOrThrow(LEFT_ARM) as ArmBodyPart,
            provider.getOrThrow(RIGHT_ARM) as ArmBodyPart
        )
    }

    public fun getLegs(provider: BodyPartProvider): Pair<LegBodyPart, LegBodyPart> {
        return Pair(
            provider.getOrThrow(LEFT_LEG) as LegBodyPart,
            provider.getOrThrow(RIGHT_LEG) as LegBodyPart
        )
    }

    public fun getFeet(provider: BodyPartProvider): Pair<FootBodyPart, FootBodyPart> {
        return Pair(
            provider.getOrThrow(LEFT_FOOT) as FootBodyPart,
            provider.getOrThrow(RIGHT_FOOT) as FootBodyPart
        )
    }
}
